### PR TITLE
Update Callbacks Documentation: Specify 'after_commit' and 'after_rollback' Behavior for Multiple Objects of the Same Record [ci skip]

### PR DESCRIPTION
**Pull request by dhavalsingh**
… multiple instances of the same model

### Motivation / Background

This Pull Request has been created because this is a pretty nuanced case that can come up in complex applications. We had a bug due to this and I couldn't find any other source specifying this behavior. The rails guides should be clear on this case so folks can avoid this and be mindful of this behavior when designing or maintaining systems.

### Detail

This Pull Request changes the AR Callback Docs.

### Additional information

https://github.com/rails/rails/issues/39400 This is the only place that references this.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by dhavalsingh** [comment]
Can someone check why the buildkite check is failing? or re-run it. I didn't put the [ci skip] tag at first and then edited it. 
Should i raise another PR?

File: guides/source/active_record_callbacks.md [hunk 1]
```diff
@@ -554,6 +554,8 @@ WARNING. The code executed within `after_commit` or `after_rollback` callbacks i
 
 WARNING. Using both `after_create_commit` and `after_update_commit` with the same method name will only allow the last callback defined to take effect, as they both internally alias to `after_commit` which overrides previously defined callbacks with the same method name.
 
+WARNING. In the context of a single transaction, should you interact with multiple instances of the same model, `after_commit` or `after_rollback` callback behavior is designed to execute callbacks for only the first instance of the model. Subsequent instances of the same model within that transaction will _not_ trigger their own `after_commit` or `after_rollback` callback. This is a nuanced behavior that can impact the predictability and flow of callback execution, particularly in scenarios where multiple instances of the same model are undergoing changes in a single transaction.
```
Comment (nickhammond) [inline guides/source/active_record_callbacks.md:557]: Can you provide a simple example of what this would look like or how you're doing this transaction?

Comment (dhavalsingh) [reply guides/source/active_record_callbacks.md:557]: https://github.com/rails/rails/issues/39400 
It's given in this PR. 
Or. did you mean i should add an example to the docs?

Comment (nickhammond) [reply guides/source/active_record_callbacks.md:557]: I was thinking an actual code example but since it's mentioned in that PR never mind.

Comment (dhavalsingh) [reply guides/source/active_record_callbacks.md]: updated the msg to be more clear. I think multiple instances might be confusing

**Comment by dhavalsingh** [comment]
Can anyone review this? I really think this should be added in the guides as there is little to no chance the behaviour will change in the future as old apps can break.

**Comment by dhavalsingh** [comment]
Needs ActiveRecord tag @zzak , also feedback one. Thanks

**Review by adrianna-chang-shopify — COMMENTED** [review]
Thanks for the contribution! The doc changes look good to me. Can you wrap your lines at a column width of 80 characters and squash your commits please?

**Comment by dhavalsingh** [comment]
@adrianna-chang-shopify Done! Thanks for the review.

**Comment by p8** [comment]
I noticed the examples underneath the warnings related to the third warning, instead of this new warning.
So I've moved this warning beneath the examples: https://github.com/rails/rails/pull/49884

**Pull request merged**